packages/next/src/state.tsx

```diff
@@ -0,0 +1,136 @@
+"use client";
+
+import type {
+  SfCart,
+  SfCurrency,
+  SfCustomer,
+} from "@vue-storefront/unified-data-model";
+import React, {
+  type PropsWithChildren,
+  createContext,
+  useContext,
+  useRef,
+} from "react";
+import { type StoreApi, createStore, useStore } from "zustand";
+
+export interface SfStateProps {
+  currencies: SfCurrency[];
+  currency: SfCurrency;
+  locale: string;
+  locales: string[];
```

Comment:
If we would like to be super-typed - we can create an interface that contains languages in form:
```ts
interface AlokaiLocales {
  en: 'en';
}
```

so user can extend this in his `d.ts` file, and then in `setLocale` TypeScript doesn't allow you to set an language that are not defined

```suggestion
  locale: keyof AlokaiLocales;
  locales: (keyof AlokaiLocales)[];
```

Reply:
Ah yes, good catch. Mateusz Ostafil was pointing that out during RFC session.

Reply:
Added